indico/modules/events/registration/util.py

```diff
@@ -460,6 +463,11 @@ def modify_registration(registration, data, management=False, notify_user=True):
             # keep current value
             continue
 
+        if field_impl.is_file_field:
+            if file_data := data_by_field.get(form_item.id):
+                eta = now_utc() + timedelta(minutes=10)
+                delete_previous_registration_file.apply_async([file_data, file_data.storage_backend,
```

Comment:
- `.delay()` is a nicer API than `.apply_async()` and it's what we usually use to call a task
- `countdown=600` is nicer here than `eta` since you don't need to calculate the end time yourself

Reply:
Please what is the way to call .delay() and pass the countdown to it? The only example I see that used countdown is in `send_email_task` which involves some techniques for sending a task again if it failed

Reply:
Ah indeed, let me check again...

Reply:
You're right, `.delay()` does not support execution options. This would work though:

```py
            delete_previous_registration_file.s(
                file_data, file_data.storage_backend, file_data.storage_file_id
            ).apply_async(countdown=600)
```

But I think it's fine to leave it as-is with `apply_async`.